src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -753,11 +753,41 @@ public void stateChanged(ChangeEvent e)   {
         }
     }
 
-    // PENDING(hmuller) - the next three methods should be removed
+    /**
+     * Unlike most components, {@code JScrollBar} derives the minimum size from
+     * the preferred size in one axis and a fixed minimum size in the other.
+     * Thus, it overrides {@code JComponent.setMinimumSize} contract
+     * that subsequent calls to {@code getMinimumSize} will return the
+     * same value as set in {@code JComponent.setMinimumSize}.
+     *
+     * @param minimumSize the new minimum size of this component
+     */
+    public void setMinimumSize(Dimension minimumSize) {
+        super.setMinimumSize(minimumSize);
+    }
 
     /**
-     * The scrollbar is flexible along it's scrolling axis and
+     * Unlike most components, {@code JScrollBar} derives the maximum size from
+     * the preferred size in one axis and a fixed maximum size in the other.
+     * Thus, it overrides {@code JComponent.setMaximumSize} contract
+     * that subsequent calls to {@code getMaximumSize} will return the
+     * same value as set in {@code JComponent.setMaximumSize}.
+     *
+     * @param maximumSize the desired maximum allowable size
+     */
+    public void setMaximumSize(Dimension maximumSize) {
+        super.setMaximumSize(maximumSize);
+    }
+
+    /**
+     * Returns the minimum size for the {@code JScrollBar}.
+     * The scrollbar is flexible along its scrolling axis and
      * rigid along the other axis.
+     * As specified in {@code setMinimumSize} JScrollBar will derive the
+     * minimum size from the preferred size in one axis and a
+     * fixed minimum size in the other.
+     *
+     * @return the minimum size as specified above.
```

Comment:
```suggestion
     * @return the minimum size as specified above
```
`@return` tags usually don't have a period in the end. Let's follow it for consistency.